test/register-bson.js

```diff
@@ -1,7 +1,65 @@
 'use strict';
 
-// TODO(NODE-3555): When we remove karma in favor of vm we can use ts-node to test src instead of lib here.
-const BSON = require('../lib/bson');
-const { ensureBuffer } = require('../lib/ensure_buffer');
-BSON.ensureBuffer = ensureBuffer;
+require('source-map-support').install({
+  hookRequire: true
+});
+
+const { inspect } = require('node:util');
+const vm = require('node:vm');
+const fs = require('node:fs');
+const path = require('node:path');
+const chai = require('chai');
+
+/**
+ * In the runInContext "web" testing instanceof checks fail
+ * since the error classes are declared in a different realm (?)
+ * So here we augment the chai assertion to fallback on checking the name property of the error instance
+ */
+chai.use(function (chai) {
+  const throwsAssertion = chai.Assertion.prototype.throw;
+  chai.Assertion.addMethod('throw', function (...args) {
+    try {
+      throwsAssertion.call(this, ...args);
+    } catch (assertionError) {
+      if (assertionError.actual?.name === assertionError.expected) {
+        return;
+      }
+      throw assertionError;
+    }
+  });
+});
+
+// Register expect globally, this is no longer necessary since we do not use karma / bundle everything together
+globalThis.expect = chai.expect;
+
+// Controls whether to run BSON library declaration in an node or "web" environment
+const web = process.env.WEB === 'true';
+console.error(inspect({ web }, { colors: true }));
+
+// Node.js 16 was when the atob and btoa globals were introduced, so we need replacements for testing on 14
+const shim_btoa = input => Buffer.prototype.toString.call(Buffer.from(input), 'base64');
+const shim_atob = input => Buffer.from(input, 'base64').toString('binary');
+
+let BSON;
+if (web) {
+  // TODO(NODE-4713): Using the umd for now since it works well as a Node.js import
+  // Switch to the .cjs rollup planned for NODE-4713
+  const filename = path.resolve(__dirname, '../dist/bson.browser.umd.js');
+  const code = fs.readFileSync(filename, { encoding: 'utf8' });
+  // These are the only globals BSON strictly depends on
+  // an optional global is crypto
```

Comment:
Did not include crypto b/c our BSON library works without it (falling back to Math.random()), but this could be an interesting way to test that fallback explicitly (we are now since I don't include crypto)